File: level_1/matrix_inplace_sorting/langs/compiler.java

class Main
{
    public static void main(String[] args)
    {
        Scanner scan = new Scanner(System.in);
        int m;
        int n;
        m = scan.nextInt();
        n = scan.nextInt();
        int input[][] = new int[m][n];

        boolean invalid = false;
        scan.nextLine();
        for (int i = 0; i < m; i++)
        {
            String row = scan.nextLine();
            String tokens[] = row.split(" ");
            int num_elements_in_a_row = 0;
            // Check if the string/row is empty
            if (row == "")
            {
                invalid = true;
                break;
            }
            for (String val : tokens)
            {
                input[i][num_elements_in_a_row] = Integer.parseInt(val);
                num_elements_in_a_row++;
            }
            // If any row has lesser/greater number of elements than n, the input is invalid
            if (num_elements_in_a_row != n)
                invalid = true;
        }

        // Check if the input is valid
        if (invalid)
        {
            System.out.println("You have entered an invalid input!");
            return;
        }
        Test test = new Test();
        test.sortMatrix(input, m, n);
        // Print the sorted matrix
        for (int i = 0; i < m; i++)
        {
            for (int j = 0; j < n; j++)
            {
                System.out.print(input[i][j] + " ");
            }
            System.out.println();
        }
    }
}

File: level_1/matrix_inplace_sorting/langs/compiler.py
def main():
    m = int(input())
    n = int(input())

    input_arr = [[0 for i in range(n)] for i in range(m)]
    invalid = False

    for i in range(m):
        num_elements_in_a_row = 0
        row = input()
        # Check if the string/row is empty
        if len(row) == 0:
            return
        tokens = row.split(' ')
        for j, val in enumerate(tokens):
            input_arr[i][j] = int(val)
            num_elements_in_a_row += 1

        # If any row has lesser/greater number of elements than n, the input is invalid
        if num_elements_in_a_row != n:
            invalid = True
            return

    # Check if the input is valid
    if invalid:
        print("You have entered an invalid input!")
        return

    sortMatrix(input_arr, m, n)
    # Print the sorted matrix
    for i in range(m):
        for j in range(n):
            print(input_arr[i][j], ' ', sep='', end='')
        print()

if __name__ == '__main__':
    main()